src/app/data-client.ts

```diff
@@ -27,6 +27,50 @@ export class DataClient {
     this.service = new DataServiceClient(serviceHost, grpcOptions);
   }
 
+  /**
+   * Obtain unified tabular data and metadata, queried with SQL.
+   *
+   * @param organizationId The ID of the organization that owns the data
+   * @param query The SQL query to run
+   * @returns An array of data objects
+   */
+  async tabularDataBySQL(organizationId: string, query: string) {
+    const { service } = this;
+
+    const req = new pb.TabularDataBySQLRequest();
+    req.setOrganizationId(organizationId);
+    req.setSqlQuery(query);
+
+    const response = await promisify<
+      pb.TabularDataBySQLRequest,
+      pb.TabularDataBySQLResponse
+    >(service.tabularDataBySQL.bind(service), req);
+    const dataList = response.getDataList();
+    return dataList.map((struct) => struct.toJavaScript());
+  }
+
+  /**
+   * Obtain unified tabular data and metadata, queried with MQL.
+   *
+   * @param organizationId The ID of the organization that owns the data
+   * @param query The MQL query to run
+   * @returns An array of data objects
+   */
+  async tabularDataByMQL(organizationId: string, query: string) {
+    const { service } = this;
+
+    const req = new pb.TabularDataByMQLRequest();
+    req.setOrganizationId(organizationId);
+    req.setMqlQuery(query);
+
+    const response = await promisify<
+      pb.TabularDataByMQLRequest,
+      pb.TabularDataByMQLResponse
+    >(service.tabularDataByMQL.bind(service), req);
+    const dataList = response.getDataList();
+    return dataList.map((struct) => struct.toJavaScript());
```

Comment:
Convenient `toJavaScript` helper! I see that this is already used below in `tabularDataByFilter` and just verifying that it was already tested on real data (either already for `tabularDataByFilter` or for this new API)

Reply:
I tested it using `select * from readings limit 5` from the ticket and received

```json
[
  {
    "readings": {
      "additional_parameters": {},
      "component_name": "fake-mov-new-actual",
      "component_type": "rdk:component:movement_sensor",
      "data": {
        "value": 25
      },
      "location_id": "5qtyfg8nb8",
      "method_name": "CompassHeading",
      "organization_id": "7f0ee750-1aa7-4e34-8a57-d83d1520ef63",
      "part_id": "b05fd0ab-d4b3-4131-ab8b-38730a12033d",
      "robot_id": "7f234851-bd33-4613-8f50-effcf1d6ac97",
      "tags": null,
      "time_received": "2023-11-09 17:44:05.614 +0000 UTC",
      "time_requested": "2023-11-09 17:44:05.614 +0000 UTC"
    }
  },
  {
    "readings": {
      "additional_parameters": {},
      "component_name": "fake-pwr-new-actual",
      "component_type": "rdk:component:power_sensor",
      "data": {
        "watts": 9.8
      },
      "location_id": "5qtyfg8nb8",
      "method_name": "Power",
      "organization_id": "7f0ee750-1aa7-4e34-8a57-d83d1520ef63",
      "part_id": "b05fd0ab-d4b3-4131-ab8b-38730a12033d",
      "robot_id": "7f234851-bd33-4613-8f50-effcf1d6ac97",
      "tags": null,
      "time_received": "2023-11-09 17:44:05.616 +0000 UTC",
      "time_requested": "2023-11-09 17:44:05.616 +0000 UTC"
    }
  },
  {
    "readings": {
      "additional_parameters": {},
      "component_name": "fake-arm-new-actual",
      "component_type": "rdk:component:arm",
      "data": {
        "pose": {
          "o_x": 0,
          "o_y": 0,
          "o_z": 1,
          "theta": 0,
          "x": 0,
          "y": 0,
          "z": 0
        }
      },
      "location_id": "5qtyfg8nb8",
      "method_name": "EndPosition",
      "organization_id": "7f0ee750-1aa7-4e34-8a57-d83d1520ef63",
      "part_id": "b05fd0ab-d4b3-4131-ab8b-38730a12033d",
      "robot_id": "7f234851-bd33-4613-8f50-effcf1d6ac97",
      "tags": null,
      "time_received": "2023-11-09 17:43:55.62 +0000 UTC",
      "time_requested": "2023-11-09 17:43:55.62 +0000 UTC"
    }
  },
  {
    "readings": {
      "additional_parameters": {},
      "component_name": "fake-arm-new-actual",
      "component_type": "rdk:component:arm",
      "data": {
        "pose": {
          "o_x": 0,
          "o_y": 0,
          "o_z": 1,
          "theta": 0,
          "x": 0,
          "y": 0,
          "z": 0
        }
      },
      "location_id": "5qtyfg8nb8",
      "method_name": "EndPosition",
      "organization_id": "7f0ee750-1aa7-4e34-8a57-d83d1520ef63",
      "part_id": "b05fd0ab-d4b3-4131-ab8b-38730a12033d",
      "robot_id": "7f234851-bd33-4613-8f50-effcf1d6ac97",
      "tags": null,
      "time_received": "2023-11-09 17:44:05.62 +0000 UTC",
      "time_requested": "2023-11-09 17:44:05.62 +0000 UTC"
    }
  },
  {
    "readings": {
      "additional_parameters": {},
      "component_name": "fake-gantry-new",
      "component_type": "rdk:component:gantry",
      "data": {
        "lengths_mm": [
          5
        ]
      },
      "location_id": "5qtyfg8nb8",
      "method_name": "Lengths",
      "organization_id": "7f0ee750-1aa7-4e34-8a57-d83d1520ef63",
      "part_id": "b05fd0ab-d4b3-4131-ab8b-38730a12033d",
      "robot_id": "7f234851-bd33-4613-8f50-effcf1d6ac97",
      "tags": null,
      "time_received": "2023-11-10 22:50:57.585 +0000 UTC",
      "time_requested": "2023-11-10 22:50:57.585 +0000 UTC"
    }
  }
]
```

Reply:
Amazing!